tools/clang/lib/Sema/SemaDecl.cpp

```diff
@@ -14629,3 +14630,21 @@ AvailabilityResult Sema::getCurContextAvailability() const {
   return D ? D->getAvailability() : AR_Available;
 }
 
+// HLSL Change Begin
+void Sema::DiagnoseSemanticDecl(hlsl::SemanticDecl *Decl) {
+  StringRef SemName = Decl->SemanticName;
+
+  StringRef BaseSemName; // The 'FOO' in 'FOO1'
+  uint32_t SemIndex;     // The '1' in 'FOO1'
+
+  // Split name and index.
+  hlsl::Semantic::DecomposeNameAndIndex(SemName, &BaseSemName, &SemIndex);
+
+  // The valid semantic indices for SV_Target[n] are 0 <= n <= 7.
```

Comment:
nit: 7 is a bit magic.

It would be nice to have a named definition for this.
But it's not going to change any time soon.  I'm happy if Chris is happy.

Reply:
I think this is fine. 7 is a bit magic, but I tend to find named definitions hurt code readability, and there's really no way this changes anytime soon.